### PR TITLE
UNR-948: add support for passing in command line flags for local launch

### PR DESCRIPTION
**Pull request by ieva-improbable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This change allows passing in command line flags for `spatial local launch` by specifying them in _Project Settings_ under  _SpatialOS GDK for Unreal_:

image

Flags can be added both with and without `--`, and get appended to the end of launch command. For instance, adding the flags in the above example would resolve to executing:
`spatial.exe local launch DefaultLaunchConfig.json --runtime_log_level debug --enable_jvm_debugging --enable_pre_run_check --experimental_runtime someruntime --snapshot snappy  ^& pause`.

#### Release note

"Feature:" enable passing in command line flags for local launch via _Project Settings_.

#### Tests

Locate to _SpatialOS GDK for Unreal_ settings under _Project Settings_ and add some flags under _Command line flags for local launch_. Then start Spatial runtime by clicking on the green _Start_ button. 
Observe that the terminal window that opens up shows the flags being passed in in its header, and that the flags impact the launch (for instance, passing in an invalid flag would cause errors). You can also see the flags added to the launch command logged in the editor logs.

### CODE REVIEW

**Comment by Vatyx** [comment]
@ieva-improbable is this ready to review?

**Comment by ieva-improbable** [comment]
@Vatyx yup, it's ready to be looked at.

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h [hunk 1]
```diff
@@ -95,6 +99,19 @@ class SPATIALGDKEDITOR_API USpatialGDKEditorSettings : public UObject
 			? FPaths::ConvertRelativePathToFull(FPaths::Combine(GetSpatialOSDirectory(), FString(TEXT("schema/unreal/generated/"))))
 			: GeneratedSchemaOutputFolder.Path;
 	}
-	
+
+	FORCEINLINE FString GetSpatialOSCommandLineLaunchFlags() const
+	{
+		FString CommandLineLaunchFlags = "";
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h]:
```suggestion
		FString CommandLineLaunchFlags = TEXT("");
```

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h [hunk 2]
```diff
@@ -95,6 +99,19 @@ class SPATIALGDKEDITOR_API USpatialGDKEditorSettings : public UObject
...
+
+		for (FString Flag : SpatialOSCommandLineLaunchFlags)
+		{
+			Flag = Flag.StartsWith("--") ? Flag : "--" + Flag;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h]:
```suggestion
			Flag = Flag.StartsWith(TEXT("--")) ? Flag : TEXT("--") + Flag;
```

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h [hunk 3]
```diff
@@ -95,6 +99,19 @@ class SPATIALGDKEDITOR_API USpatialGDKEditorSettings : public UObject
...
+			CommandLineLaunchFlags += Flag + " ";
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h]:
```suggestion
			CommandLineLaunchFlags += Flag + TEXT(" ");
```

**Comment by girayimprobable** [comment]
Have we made user-facing documentation for it? If not, we should make a task to document it and let @NotRobSheridan know.

**Pull request closed without merging**